src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -444,6 +454,12 @@ public bool BulkUpdateRemove(ref CompactKeyLookup key, long pageNum, int offset,
     
     public int BulkUpdateStart(Span<CompactKeyLookup> keys, Span<long> values, Span<int> offsets, out long pageNum)
     {
+        #if DEBUG
+        for (int i = 0; i < keys.Length; i++)
+        {
+            Debug.Assert(keys[i].ContainerId == -1, "keys[i].ContainerId == -1");
+        }
+        #endif
         var read = _inner.BulkUpdateStart(keys, values, offsets, out pageNum);
```

Comment:
In Debug or should we setup that for `VALIDATION` instead?

Reply:
No, that is something that I absolutely want to get triggered as early as possible.